### PR TITLE
Split the list of instruments into two columns in the extended dashboard

### PR DESCRIPTION
**Pull request by jmborr**
[](https://code.ornl.gov/sns-hfir-scse/infrastructure/web-monitor/-/issues/17)

### Testing
Build `LOCAL` environment and go to the extended dashboard page. It should distribute the three instruments into two columns and look like this:

### Work Done
- [x] dashboard template HTML now splits list of instruments into two columns

### CODE REVIEW

File: src/webmon_app/reporting/dasmon/views.py [hunk 1]
```diff
@@ -36,8 +36,12 @@ def dashboard(request):
     """
     # Get the system health status
     global_status_url = reverse(settings.LANDING_VIEW, args=[])
+    instrument_data = view_util.get_instrument_status_summary()
+    cutoff = int(len(instrument_data) / 2)
```
Comment (wdzhou) [inline src/webmon_app/reporting/dasmon/views.py:40]: In python 3, I think `//` is for integer division.

File: src/webmon_app/reporting/dasmon/views.py [hunk 2]
```diff
@@ -36,8 +36,12 @@ def dashboard(request):
...
     template_values = {
-        "instruments": view_util.get_instrument_status_summary(),
+        "instruments": instrument_data,
```
Comment (wdzhou) [inline src/webmon_app/reporting/dasmon/views.py:42]: It is better to add some documentation here.

**Review by wdzhou — COMMENTED** [review]
Besides the in-line comment, one more general question: is there any way to create a test to reflect the requirement?

File: src/webmon_app/reporting/dasmon/views.py (hunk 1 above)
Comment (jmborr) [reply src/webmon_app/reporting/dasmon/views.py:40]: yes, but I like the more explicit casting to `int`. You rarely see the `//` operator these days

File: src/webmon_app/reporting/dasmon/views.py (hunk 2 above)
Comment (jmborr) [reply src/webmon_app/reporting/dasmon/views.py:42]: do you mean a comment in the code explaining the intent of the different dictionary keys?

Comment (jmborr) [reply src/webmon_app/reporting/dasmon/views.py:42]: > Besides the in-line comment, one more general question: is there any way to create a test to reflect the requirement?

The only check that comes to mind if to verify that function `dashboard()` can populate template `dahsboard.html` if appropriate template_values are passed on.
I see there are already some tests under `tests/` that check the Views of some pages so I could take a look to these and see if I can come up with a similar View test for the "extended dashboard" page. Does that sound good?

**Comment by jmborr** [comment]
> Besides the in-line comment, one more general question: is there any way to create a test to reflect the requirement?

> Besides the in-line comment, one more general question: is there any way to create a test to reflect the requirement?

The only check that comes to mind if to verify that function `dashboard()` can populate template `dahsboard.html` if appropriate template_values are passed on.
I see there are already some tests under `tests/` that check the Views of some pages so I could take a look to these and see if I can come up with a similar View test for the "extended dashboard" page. Does that sound good?

File: src/webmon_app/reporting/dasmon/views.py (hunk 2 above)
Comment (wdzhou) [reply src/webmon_app/reporting/dasmon/views.py:42]: A comment such that the people in the future can easily understand that the instruments are put to 2 columns.

File: tests/test_fixtures.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+def test_request_page(request_page):
+    response = request_page("/dasmon/", settings.GENERAL_USER_USERNAME, settings.GENERAL_USER_PASSWORD)
+    assert response.ok
+
+
+def test_HTTPText(HTTPText):
+    text = HTTPText("on and on")
+    assert "on" in text
+    assert "ona" in text
```
Comment (wdzhou) [inline tests/test_fixtures.py:14]: where does `ona` come from?

Comment (jmborr) [reply tests/test_fixtures.py:14]: [Here's the explanation](https://github.com/neutrons/data_workflow/blob/double_column/tests/conftest.py#L53):
```python
        def __contains__(self, substring: str):
            r"""check HTML string contains a substring, but whitespaces are not considered
            :param substring: substring to match
            """
```

**Review by wdzhou — APPROVED** [review]
Looks good to me

:shipit:

**Comment by jmborr** [comment]
💃 thank you!

**Pull request merged**